src/main/java/graph/LocustBarChart.java

```diff
@@ -0,0 +1,93 @@
+        JFreeChart chart = ChartFactory.createBarChart(TITLE,CATEGORYAXISLABELTITLE,VALUEAXISLABELTITLE, createDataset(locustReportReader, testResultsIteration), PlotOrientation.VERTICAL,false,true,false);
+        chart.addSubtitle(0, new TextTitle(" "));
+        chart.addSubtitle(1, new TextTitle(this.createRequestResults(locustReportReader, testResultsIteration)));
+        chart.addSubtitle(2, new TextTitle(" "));
+        try {
+            ChartUtils.saveChartAsPNG(file, chart, CHARTWIDTH, CHARTHEIGHT);
+        } catch (IOException error) {
+            System.out.println(error.getMessage());
+            System.out.println(ERROR);
+        }
+    }
+	
+	/**
+	 * Creates the dataset for the distribution of load
+	 * @param locustReportReader the instance of the csv reader
+	 * @param testResultsIteration the index of the csv results in case of the csv contains more than one line results
+	 * @return it returns the dataset to be used in the public createChart method
+	 */
+    private CategoryDataset createDataset(LocustReportReader locustReportReader, int testResultsIteration) {
+        DefaultCategoryDataset dataset = new DefaultCategoryDataset();
+        dataset.setValue(Integer.parseInt(locustReportReader.getLocustDistributionDataList().get(testResultsIteration).get50()), "Requests", "50%");
+        dataset.setValue(Integer.parseInt(locustReportReader.getLocustDistributionDataList().get(testResultsIteration).get66()), "Requests", "66%");
+        dataset.setValue(Integer.parseInt(locustReportReader.getLocustDistributionDataList().get(testResultsIteration).get75()), "Requests", "75%");
+        dataset.setValue(Integer.parseInt(locustReportReader.getLocustDistributionDataList().get(testResultsIteration).get80()), "Requests", "80%");
+        dataset.setValue(Integer.parseInt(locustReportReader.getLocustDistributionDataList().get(testResultsIteration).get90()), "Requests", "90%");
+        dataset.setValue(Integer.parseInt(locustReportReader.getLocustDistributionDataList().get(testResultsIteration).get95()), "Requests", "95%");
+        dataset.setValue(Integer.parseInt(locustReportReader.getLocustDistributionDataList().get(testResultsIteration).get98()), "Requests", "98%");
+        dataset.setValue(Integer.parseInt(locustReportReader.getLocustDistributionDataList().get(testResultsIteration).get99()), "Requests", "99%");
+        dataset.setValue(Integer.parseInt(locustReportReader.getLocustDistributionDataList().get(testResultsIteration).get100()), "Requests", "100%");
+        return dataset;
+    }
+    
+    /**
+     * It generate the information of the performance results
+     * @param locustReportReader the instance of the csv reader
+     * @param testResultsIteration the index of the csv results in case of the csv contains more than one line results
+     */
+    private String createRequestResults(LocustReportReader locustReportReader, int testResultsIteration) {
```

Comment:
Where is this used?

Reply:
This is used in the line 43 in the method createChart(...). It creates the string line with the request information that appears on top of the image.